e2e/scenario_test.go

```diff
@@ -921,6 +921,30 @@ func Test_Ubuntu2204_WASM(t *testing.T) {
 	})
 }
 
+func Test_AzureLinux_Skip_Binary_Cleanup(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "tests that an AzureLinux node will skip binary cleanup and can be properly bootstrapped",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDAzureLinuxV2Gen2,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				if nbc.KubeletConfig == nil {
```

Comment:
unless we want to actually add something to the kubelet config here, I don't think we need this mutator